randovania/resolver/resolver_reach.py

```diff
@@ -138,25 +136,21 @@ def calculate_reach(cls, logic: Logic, initial_state: State) -> ResolverReach:
                         requirement = RequirementAnd([requirement, requirement_to_leave])
 
                 # Check if the normal requirements to reach that node is satisfied
-                satisfied = requirement_including_leaving.satisfied(initial_state.resources, energy, database)
+                satisfied = requirement_including_leaving.satisfied(context, energy)
                 if satisfied:
                     # If it is, check if we additional requirements figured out by backtracking is satisfied
-                    satisfied = logic.get_additional_requirements(node).satisfied(
-                        initial_state.resources, energy, initial_state.resource_database
-                    )
+                    satisfied = logic.get_additional_requirements(node).satisfied(context, energy)
```

Comment:
Add `assert node is not None`

Reply:
Nearly the same here: If assertion is too time consuming, could still add a `# type: ignore`